src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs

```diff
@@ -16,6 +16,208 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+        void OnCompleted(System.Action action);
+    }
+}
+
+namespace System.Threading.Tasks {
+    public class Awaiter<T>: System.Runtime.CompilerServices.INotifyCompletion {
+        public bool IsCompleted  => true;
+        public void OnCompleted(System.Action action) {}
+        public T GetResult() { throw new Exception(); }
+    }
+    public class Task<T> {}
+}
+";
+            var extensionSource = @"
+    public static class MyExtensions {
+        public static System.Threading.Tasks.Awaiter<T> GetAwaiter<T>(this System.Threading.Tasks.Task<T> task) {
+            return null;
+        }
+    }
+";
+
+            if (includeNamespaceAroundTaskExtension)
+            {
+                corAssembly = corAssembly + $"\nnamespace Hidden {{\n{extensionSource}\n }}";
+            }
+            else
+            {
+                corAssembly = corAssembly + "\n" + extensionSource;
+            }
+
+
+            var corCompilation = CreateCompilation(corAssembly, references: new[] { MscorlibRef_v20 });
+            corCompilation.VerifyDiagnostics();
+            return corCompilation.ToMetadataReference();
+        }
+
+        [WorkItem(19048, "https://github.com/dotnet/roslyn/pull/19623")]
+        [Fact]
+        public void NonStandardTaskImplementation_NoGlobalUsing_NoScriptUsing()
+        {
+
+            var script = CreateCompilation(
+                source: @" System.Console.Write(""complete"");",
+                parseOptions: TestOptions.Script,
+                options: TestOptions.DebugExe,
+                references: new MetadataReference[] { TaskFacadeAssembly(), MscorlibRef_v20 });
+
+            script.VerifyEmitDiagnostics(
+                // error CS1061: 'Task<object>' does not contain a definition for 'GetAwaiter' and no extension method 'GetAwaiter' accepting a first argument of type 'Task<object>' could be found (are you missing a using directive or an assembly reference?)
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "").WithArguments("System.Threading.Tasks.Task<object>", "GetAwaiter").WithLocation(1, 1));
+
+        }
+
+        [WorkItem(19048, "https://github.com/dotnet/roslyn/pull/19623")]
+        [Fact]
+        public void NonStandardTaskImplementation_NoGlobalUsing_NoScriptUsing_NoNamespace()
+        {
+
+            var script = CreateCompilation(
+                source: @" System.Console.Write(""complete"");",
+                parseOptions: TestOptions.Script,
+                options: TestOptions.DebugExe,
+                references: new MetadataReference[] { TaskFacadeAssembly(false), MscorlibRef_v20 });
+
+            script.VerifyEmitDiagnostics();
+
+        }
+
+
+        [WorkItem(19048, "https://github.com/dotnet/roslyn/pull/19623")]
+        [Fact]
+        public void NonStandardTaskImplementation_GlobalUsing_NoScriptUsing_VoidHidden()
+        {
+            var script = CreateCompilation(
+                source: @"class C {}",
```

Comment:
>class C {} [](start = 26, length = 10)

To stress the scenario more, I would not declare a class in the script, I would declare an interface because it doesn't need object or void.